demos-europe/demosplan-core
# feat: further improve branding logo upload

Ticket: [DPLAN-11707](https://demoseurope.youtrack.cloud/issue/DPLAN-11707/Prod-DiPlanBeteiligung-BB-Konfiguration-nicht-moglich)

As a followup of https://github.com/demos-europe/demosplan-core/pull/2413 this PR improves UX in the branding logo upload. 

- use a button instead of a checkbox to delete the current logo. for this to work, a distinct method is introduced. To prevent unwanted deletion, a confirm dialog is implemented. the "isLogoDeletable" variable could be removed, then.
- the newly uploaded image is not set as branding.logoHash on setFile (because at this time, it is not yet saved) but only on after saveBranding. at this point, the method clearFilesList from DpUploadFiles is also called to remove the previously uploaded file.

### How to review/test
- already uploaded logo should be deletable via button, deletion has to be confirmed
- when selecting a new file to upload and saving it, the files list should be emptied. the newly uploaded image should only appear in the "Aktuelles Logo" spot after having saved, not after the upload itself has finished

### PR Checklist
